tarantool/connection.py

```diff
@@ -767,12 +769,30 @@ def _send_request_wo_reconnect(self, request):
 
         assert isinstance(request, Request)
 
+        # Flag for detecting the last message as out-of-band.
+        iproto_chunk_detected = False
+        
         response = None
         while True:
             try:
-                self._socket.sendall(bytes(request))
+                if not iproto_chunk_detected:
+                    # If the last received message is out-of-band, 
+                    # the request will not be sent.
+                    self._socket.sendall(bytes(request))
                 response = request.response_class(self, self._read_response())
-                break
+                if response._code == IPROTO_CHUNK:
+                    # Сase of receiving an out-of-band message.
+                    pushed_data.append(copy.deepcopy(response._data))
+                    if callable(on_push):
+                        # Callback function with data from out-of-band 
+                        # message is being called.
+                        on_push_res.append(on_push(copy.deepcopy(response._data)))
```

Comment:
Since the response for `IPROTO_CHUNK` isn't exposed anywhere else, I think it will be ok to remove copy if we get rid of `pushed_data` and `on_push` for a single `on_push(data, on_push_ctx)`.

Reply:
Done